### PR TITLE
support git@blabla in add

### PR DESCRIPTION
**Pull request by innerlee**
How to test this?

Fix https://github.com/JuliaLang/Pkg.jl/issues/321

### CODE REVIEW

**Comment by KristofferC** [comment]
Could add a test to `repl.jl` that tests something like `add git@github.com:JuliaLang/Example.jl.git`.

**Comment by pfitzseb** [comment]
It would probably be better to check that the preceding "word" is a valid package name and the `@` is followed by a valid version range, since you could have a package called `git`.

**Comment by innerlee** [comment]
So I do not need to worry about the credentials, right?

File: src/REPLMode.jl [hunk 1]
```diff
@@ -167,7 +167,7 @@ end
 ################
 # REPL parsing #
 ################
-const lex_re = r"^[\?\./\+\-](?!\-) | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
+const lex_re = r"^[\?\./\+\-](?!\-) | git@[^@\#\s;]+ | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
```
Comment (StefanKarpinski) [inline src/REPLMode.jl]: That `git@` seems very specific, but I guess it gets the job done for now.

Comment (innerlee) [reply src/REPLMode.jl]: How about [this regex](https://stackoverflow.com/a/22312124)
`((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?`

**Comment by innerlee** [comment]
Okay, so the credential thing is needed. Any workaround?

```bash
  [8dfed614] - Test 
   Cloning git-repo `git@github.com:JuliaLang/Example.jl.git`
Private key location for 'git@github.com':
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

File: src/REPLMode.jl [hunk 2]
```diff
@@ -167,7 +167,7 @@ end
 ################
 # REPL parsing #
 ################
-const lex_re = r"^[\?\./\+\-](?!\-) | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
+const lex_re = r"^[\?\./\+\-](?!\-) | ((git|ssh|http(s)?)|(git@[\w\.]+))(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)? | [^@\#\s;]+\s*=\s*[^@\#\s;]+ | \#\s*[^@\#\s;]* | @\s*[^@\#\s;]* | [^@\#\s;]+|;"x
```
Comment (KristofferC) [inline src/REPLMode.jl]: This regex is becoming quite a beast :D. Oh well.

**Comment by KristofferC** [comment]
We should probably change the REPL parsing a bit but this should be good enough for now. Thanks!

**Comment by tkoolen** [comment]
Thank you!

File: src/REPLMode.jl (hunk 2 above)
Comment (StefanKarpinski) [reply src/REPLMode.jl]: It has an `x` modifier so it can be split across lines.

Comment (omus) [reply src/REPLMode.jl]: Feel free to take inspiration from: https://github.com/JuliaLang/julia/blob/master/stdlib/LibGit2/src/utils.jl#L5-L26

**Pull request merged**